sat/cli/bootsys/mgmt_power.py

```diff
@@ -428,10 +428,71 @@ def do_power_on_ncns(args):
                         except FatalPlatformError as err:
                             LOGGER.error(f'Failed to unfreeze Ceph on storage NCNs: {err}')
                             sys.exit(1)
-                        LOGGER.info('Ceph freeze completed successfully on storage NCNs.')
+                        LOGGER.info('Ceph unfreeze completed successfully on storage NCNs.')
+
+                        # Mount Ceph and S3FS filesystems on ncn-m001
+                        try:
+                            mount_filesystems_on_ncn('ncn-m001')
+                        except MountError as err:
+                            LOGGER.error(f'Failed to mount filesystems on ncn-m001: {err}')
+                            sys.exit(1)
+                        LOGGER.info('Mounted Ceph and S3FS filesystems on ncn-m001 successfully.')
 
         except ConsoleLoggingError as err:
             LOGGER.error(f'Aborting boot of NCNs due failure to set up NCN console logging: {err}')
             raise SystemExit(1)
 
     LOGGER.info('Succeeded with {}.'.format(action_msg))
+
+
+def mount_filesystems_on_ncn(ncn):
+    """Mount Ceph and S3FS filesystems on a given NCN.
+
+    Args:
+        ncn: The hostname of the NCN where the filesystems will be mounted.
+
+    Raises:
+        MountError: If mounting any filesystem fails.
+    """
+    filesystems = {
+        "ceph": "awk '{ if ($3 == \"ceph\") { print $2; }}' /etc/fstab",
+        "fuse.s3fs": "awk '{ if ($3 == \"fuse.s3fs\") { print $2; }}' /etc/fstab"
+    }
+
+    for fs_type, awk_command in filesystems.items():
+        try:
+            result = subprocess.run(awk_command, shell=True, capture_output=True, text=True, check=True)
+            mount_points = result.stdout.splitlines()
+
+            for mount_point in mount_points:
+                result = subprocess.run(["mountpoint", mount_point], capture_output=True, text=True)
+                if result.returncode != 0:  # If not a mountpoint, try to mount it
+                    subprocess.run(["mount", mount_point], check=True)
+        except subprocess.CalledProcessError as e:
+            raise MountError(f"Failed to mount {fs_type} filesystem on {ncn}: {e.stderr}")
+
+
+class MountError(Exception):
+    """Custom exception for mount errors."""
+
+    def __init__(self, message, output=None):
+        """
+        Initialize the MountError.
+
+        Args:
+            message (str): The error message.
+            output (str, optional): Additional output or details about the error.
+        """
+        super().__init__(message)
+        self.output = output
+
+    def __str__(self):
+        """
+        Return a string representation of the error.
+
+        Returns:
+            str: The error message and additional output if available.
+        """
+        if self.output:
+            return f"{self.args[0]}\nOutput: {self.output}"
+        return self.args[0]
```

Comment:
It doesn't look like you're really using any of the custom exception handling code here. You're just passing in a `message` when you raise this type of exception. I would suggest simplifying:

```suggestion
    pass
```